### ISSUE TITLE
Network logger SSL pinning

### ISSUE THREAD

**Issue by BucekJiri** [Instabug/Instabug-iOS#342]
### The problem
How to implement SSL pinning using NetworkLogger.

### URLSession delegate usual implementation
The completion handler uses URLCredential as Optional, returning Credential with serverTrust in case of success and nil in case of failure.

```swift
    func urlSession(_ session: URLSession, didReceive challenge: URLAuthenticationChallenge, completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void) {

        guard let serverTrust = challenge.protectionSpace.serverTrust else {
            completionHandler(.performDefaultHandling, nil)
            return
        }
        if checkValidity(of: serverTrust) {
            let credential = URLCredential(trust: serverTrust)
            completionHandler(.useCredential, credential)
        } else {
            completionHandler(.cancelAuthenticationChallenge, nil)
        }
    }
```

### Network logger implementation
How to implement the same thing using NetworkLogger? Should I do the certificate validation in the `setCanAuthenticate` method and then return the URLCredential in the `setDidReceive` method? What do I return in case of failure? ServerTrust is optional. Wouldn't it make more sense to return URLCredential? as Optional? 

`setCanAuthenticateAgainstProtectionSpaceHandler(_ protectionSpaceHandler: @escaping (URLProtectionSpace) -> Bool)`

`setDidReceiveAuthenticationChallengeHandler(_ reciveChallengeHandler: @escaping (URLAuthenticationChallenge) -> URLCredential)`

**Comment by Kmohamed** [comment]
Hello @BucekJiri 
To be on the same page we want to change `URLCredential` to be `Optional` in order to handle failure. am I right?

**Comment by BucekJiri** [comment]
Yes. I believe that would solve the problem.

**Comment by Kmohamed** [comment]
I agree with you. I will make sure to add it in the next release.
Thanks @BucekJiri 🙏

**Comment by BucekJiri** [comment]
@Kmohamed Perfect. Thank you for that.

**Comment by BucekJiri** [comment]
@Kmohamed Saw two new releases but the URLCredentials are still not Optional. Do you plan to release this?

**Comment by Kmohamed** [comment]
@BucekJiri I am sorry for not releasing it. Because that release was a hotfix for an issue and will be included in the next release.

**Comment by BucekJiri** [comment]
@Kmohamed No worries. I will be watching the releases.

**Comment by Kmohamed** [comment]
Hi @BucekJiri  we Just released SDK [version 9.0.4](https://github.com/Instabug/Instabug-iOS/releases/tag/9.0.4) with the fix
